### ISSUE TITLE
Interleaved groupnorm: legacy two-pass reduction computes an inexact group mean for multi-tile groups

### ISSUE THREAD

**Issue by iwroszTT** [tenstorrent/tt-metal#53846]
### Summary

`ttnn.group_norm`'s legacy two-pass statistics path computes an **inexact group mean for most multi-tile groups**. A single tile per group is always exact; beyond that the mean is wrong at 14 of the first 20 tile counts, with the error quantized to integer multiples of a fixed step. A group whose variance is at or below `eps` therefore normalizes to about ±1 instead of about 0, because the division amplifies the mean error by `1/sqrt(eps)`.

`use_welford=True` is bit-exact at every tile count tested. `fp32` input does not help — it fails on *more* tile counts than bf16. No `MathFidelity` or `fp32_dest_acc_en` setting rescues it.

Verified on `main` @ `f2a9207b30b` (2026-08-20), Blackhole p100a, `GroupNormNoMcastProgramFactory` with `device/kernels/compute/groupnorm.cpp`, grid `{1,1}`, interleaved TILE.

CODEOWNERS: @tenstorrent/metalium-developers-mmfusedreduce

### How to reproduce

The probe is a **single constant group**: every element of the group holds the same value, so the true variance is 0 and the exact golden output is 0 for every element. That makes any error in the computed mean visible, and with a small `eps` it saturates to about ±1.

```python
# gn_repro.py  --  legacy two-pass groupnorm returns a nonzero result for a constant group
import torch, ttnn

V = 5.0                        # any constant; the group's true mean is exactly V
device = ttnn.open_device(device_id=0)

print(" tiles  rows   max|out| (golden 0)   verdict")
for tiles in range(1, 21):
    rows = tiles * 32          # tiles per group, C=32 and num_groups=1 -> block_w = 1
    x = torch.full((1, 1, rows, 32), V)
    xt = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    out = ttnn.group_norm(
        xt,
        num_groups=1,
        epsilon=1e-5,
        core_grid=ttnn.CoreGrid(y=1, x=1),
        inplace=False,          # TILE layout rejects the python default inplace=True
    )
    m = ttnn.to_torch(out).abs().max().item()
    print(f" {tiles:<6} {rows:<6} {m:.4f}                 {'ok' if m <= 0.03 else 'WRONG'}")

ttnn.close_device(device)
```

Run it with:

```bash
export TT_METAL_HOME=$(pwd)
python_env/bin/python gn_repro.py
```

To read out the **mean error itself** rather than the saturated output, change `epsilon=1e-5` to `epsilon=1.0`. That makes the division well conditioned, so the output is approximately `-(mean_error)` directly.

To confirm the numerically sound path is unaffected, add `use_welford=True` to the `group_norm` call — every tile count then returns exactly 0.

**In-tree repro.** #53677 adds the first merge-gate coverage for these factories and carries this as a disabled cell in `tests/ttnn/unit_tests/gtests/test_normalization.cpp`:

```
NormalizationSmoke.DISABLED_GroupNormMultiRowTileWideC     # [1,1,64,64], num_groups=2 -> 2 tiles per group
```

Run with `--gtest_also_run_disabled_tests`. Enable the cell when this is fixed.

### Which tile counts are wrong

Constant group, `C=32`, `num_groups=1`, bf16, `eps=1e-5`, golden 0 everywhere. Metric is max absolute output:

| tiles per group | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| legacy, bf16 | 0.0000 | **0.9922** | **0.9922** | 0.0000 | 0.0000 | **0.9922** | **0.9922** | 0.0000 | **0.9922** | **0.9648** |
| `use_welford=True` | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 |

| tiles per group | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| legacy, bf16 | **0.9922** | **0.9570** | 0.0000 | **0.9727** | **0.9766** | 0.0000 | **0.9922** | **0.9727** | **0.9648** | **0.9922** |
| `use_welford=True` | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 | 0.0000 |

Exact only at tiles-per-group in **{1, 4, 5, 8, 13, 16}**; wrong at the other 14 of the first 20.

Reproduced on a second shape (`C=64`, `num_groups=2`, two tile-aligned groups), same behaviour at the tile counts tested — 1, 4 and 5 exact; 2, 3 and 6 wrong.

### The error is quantized

Re-running with `eps=1.0` reads out the mean error directly (`V = 5.0`):

| tiles per group | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| mean error | 0.0000 | +0.031006 | -0.031006 | 0.0000 | 0.0000 | +0.031006 | -0.031006 | 0.0000 | +0.031006 | -0.062012 |
| as multiple of 0.031006 | 0 | +1 | -1 | 0 | 0 | +1 | -1 | 0 | +1 | -2 |

| tiles per group | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| mean error | +0.031006 | -0.093262 | 0.0000 | -0.123047 | -0.031006 | 0.0000 | -0.031006 | +0.154297 | -0.031006 | +0.123047 |
| as multiple of 0.031006 | +1 | -3 | 0 | -4 | -1 | 0 | -1 | +5 | -1 | +4 |

Every observed error is an integer multiple of **0.031006** (about 2 ulp of 5.0 in bf16), with a multiplier between -4 and +5 that varies irregularly with the tile count. That is the signature of a running accumulator rounding to bf16 a variable number of times — not of a single fixed mis-step, and not of smooth summation error, which would grow monotonically and would not return to bit-exact at 4, 5, 8, 13 and 16 tiles.

### fp32 and compute-config knobs do not help

Same constant-group probe, `eps=1e-5`:

| tiles per group | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|
| legacy, bf16 | 0.0000 | 0.9922 | 0.9922 | 0.0000 | 0.0000 | 0.9922 | 0.9922 | 0.0000 |
| legacy, fp32 | 0.0000 | 0.9224 | 0.9756 | 0.0000 | **0.9229** | 0.9224 | 0.9224 | **0.9224** |

fp32 fails at 5 and 8 tiles where bf16 is exact, so this is not an input-precision problem.

Compute-config sweep on `[1,1,64,64]`, `num_groups=2` (2 tiles per group), reported as `g0 / g1`:

| variant | result |
|:---|:---|
| default config | 0.9492 / 0.9922 |
| `MathFidelity.LoFi` | 0.9766 / 0.9766 |
| `MathFidelity.HiFi4` | 0.9492 / 0.9922 |
| `fp32_dest_acc_en=True` | 0.9844 / 0.9844 |
| HiFi4 + `fp32_dest_acc_en` | 0.9766 / 0.0000 |
| **`use_welford=True`** | **0.0000 / 0.0000** |

### Impact

Bounded, but a silent wrong answer rather than an error:

- A group with variance at or below `eps` normalizes to about ±1 instead of about 0. Not purely synthetic — ReLU-dead or saturated channel groups and zero-padded regions produce constant groups in trained networks.
- For well-conditioned groups the effect is small. With every group at variance exactly 1, `C=64, num_groups=2` gives max absolute error 0.008 / 0.039 / 0.008 / 0.070 / 0.008 at 32 / 64 / 128 / 256 / 512 rows.

Note those well-conditioned errors do **not** track the exact/inexact split above (`tiles=8` is bit-exact on the mean, yet is the worst of that row). So the tile-count pattern describes when the group *mean* is bit-exact; it is not a complete model of the path's total error. A fix should be validated against both probes.

### Not yet localized

The mechanism is not pinned down. Two hypotheses were tested and both failed:

- **Period-4 blocking / tail handling.** Fits tiles 1-8 exactly, but breaks at 9, 12, 17 and 20 (16 of 20). Also, the `ReduceDim::REDUCE_SCALAR` path in `ttnn/cpp/ttnn/kernel_lib/reduce_helpers_compute.inl:399-473` accumulates all `Ht*Wt` tiles into a single DST register with no chunking and no tail, so there is no leftover block there to mishandle.
- **`num_out_blocks` splitting.** Self-contradictory: a single reduce over 2 tiles is inexact (`tiles=2`, `num_out_blocks=1`) while two reduces over 2 tiles each is exact (`tiles=4`, `num_out_blocks=2`).

The reduce scaler is `1/reduce_factor_w` with `reduce_factor_w = num_rows_per_batch_per_core * num_channels_per_group` (`groupnorm_no_mcast_program_factory.cpp:709`), prepared by `calculate_and_prepare_reduce_scaler<..., PoolType::AVG, REDUCE_SCALAR, reduce_factor_w>()` in `writer_unary_gn_rm_gb.cpp:264`. Scaler exactness does not explain the pattern either: `1/2048` is exactly representable yet `tiles=2` is wrong, while `1/5120` is not representable yet `tiles=5` is exact.

**Suggested next step:** `DPRINT` the contents of `dfb_ex_partial` per out-block (`kernels/compute/groupnorm.cpp:397`) and of `dfb_ex_global` after `wait_front(1)` (`:443`), on the constant-group probe at `tiles=2` (inexact) versus `tiles=4` (exact). That separates "the error enters in the per-out-block partial" from "the error enters in the global combine", which is the one fact that distinguishes the remaining possibilities. Only the JIT kernel recompiles, so the loop is fast.

### Merge Gate coverage

Nightly's interleaved groupnorm coverage is ROW_MAJOR-input only, so no existing test reaches interleaved TILE groupnorm at all.

Two corrections to `DISABLED_GroupNormMultiRowTileWideC`'s current comment, worth folding in when the cell is enabled:

1. Its stated diagnosis — "wrong results in the SECOND channel tile" — does not hold. With *all* groups made constant, channel tile 0 fails identically (0.9492). The apparent tile-specificity came from the test input making only channels 32-63 constant.
2. "whenever H\*W spans more than one row tile" is not the condition: 4, 5, 8, 13 and 16 row tiles are all bit-exact. The cell would be much stronger sweeping tiles-per-group over 1..20 as the script above does, rather than testing 2 row tiles alone.

### Related issues

- #53803 — **sub-tile group widths**: the same legacy two-pass path, but a different trigger and a different scaling. It reproduces at *one* tile per group (a value that is exact here) and degrades with group *count*, reaching 1.73 absolute error on variance-1 input. Same Welford mitigation, likely a different fix (mask application vs the reduction), so tracked apart.
- #51231 item 7 — `block_wt_last` declared `bool` in the **sharded** factory; different factory, different mechanism.
